server/application/application.go

```diff
@@ -1446,6 +1446,10 @@ func (s *Server) GetResource(ctx context.Context, q *application.ApplicationReso
 	}
 	obj, err := s.kubectl.GetResource(ctx, config, res.GroupKindVersion(), res.Name, res.Namespace)
 	if err != nil {
+		// If the resource doesn't exist yet (e.g., new resource), provide a more helpful error
+		if apierrors.IsNotFound(err) && res.UID == "" {
+			return nil, status.Errorf(codes.NotFound, "Resource %s %s/%s does not exist yet. It may be a new resource that hasn't been created.", res.Kind, res.Namespace, res.Name)
```

Comment:
Why would this be an error at all? A resource not yet existing is an expected scenario, and being able to inspect the new resource before sync'ing is a very useful feature of argocd.